gokitmiddlewares/stalemiddleware/middleware.go

```diff
@@ -0,0 +1,39 @@
+package stalemiddleware
+
+import (
+	"context"
+	"time"
+
+	"github.com/arquivei/foundationkit/app"
+	"github.com/go-kit/kit/endpoint"
+)
+
+func New(c Config) endpoint.Middleware {
+	ch := make(chan struct{})
+	go func() {
+		for {
+			select {
+			case <-ch:
+				if !app.IsHealthy() {
+					app.SetHealthy()
+				}
+			case <-time.After(c.Timeout):
+				if app.IsHealthy() {
+					if c.Logger != nil {
```

Comment:
Nit: it might be interesting in the future to use a slice of callbacks instead, as this same check can be used to manage resources or trigger background tasks when the system is idle.